packages/@aws-cdk/aws-iam/lib/oidc-provider.ts

```diff
@@ -158,7 +149,7 @@ export class OpenIdConnectProvider extends Resource implements IOpenIdConnectPro
     });
 
     this.openIdConnectProviderArn = Token.asString(resource.ref);
-    this.openIdConnectProviderIssuer = Fn.select(1, Fn.split('oidc-provider/', this.openIdConnectProviderArn));
+    this.openIdConnectProviderIssuer = Arn.extractResourceName(this.openIdConnectProviderArn, 'oidc-provider');
```

Comment:
A bit worried that this method is going to go undiscovered when someone needs it. 
Anything we can do to make this more discoverable?

I've never actually used the `Arn` class but usually use `stack.parseArn()`.
How about adding it alongside `parseArn()` in the `Stack` class?

Reply:
I'm 100% against `stack.parseArn()` in the first place. It's a member function for no good reason, since it doesn't use any member of `stack`.

So I'm even more against polluting the interface of `Stack` even further by adding more of these member functions that should have been static utilities.

Reply:
Could you then mark the one in stack as deprecated and recommend the use of the one in `Arn`?

Reply:
Honestly I think the better solution is to deprecate `stack.parseArn()` and introduce a new `Arn.parseArn()` that takes an `options` struct, which can have the functionality of `extractResourceName` integrated into as well.

I didn't do that before because I wanted this to be quick and simple and not mess too much with existing code, but I guess we're past the point of a quick fix now and we're going to clean house after all.

I thought you told me yesterday to do that change in a different PR. Should I interpret your comment as you wanting it in this PR anyway?

Reply:
Separate PR is fine.